### PR TITLE
[develop] Fix failures on AQM and verification tasks and change structure of module files on WCOSS2

### LINKED ISSUE #785: Each task should load only necessary modules on wcoss2
## Description
- Currently, multiple tasks share a module file in the app for convenience. This means that some tasks load unnecessary modules. However, NCO requires that each task load the modules necessary only for itself on WCOSS2. 
- The version control file `run_ver` needs to be updated. When running tasks, `build_ver` should not be sourced.

## Solution
- Do not use the module files used when compiling.
- Put the exact module list directly into `modulefiles/tasks/wcoss2/[task]`.
- Update `run_ver`.

### LINKED ISSUE #789: AQM and verification tasks do not work on wcoss2
## Current behavior
- AQM tasks have not worked on wcoss2 since PR #761 was merged. 
- The verification tasks have not worked correctly on wcoss2 since PR #676 was merged.

## Machines affected
WCOSS2

## Detailed Description of Fix
- AQM tasks: update module files in `modulefiles/tasks/wcoss2/[task]`.
- Verification tasks: remove `wcoss2` from if-statement in `parm/wflow/verify.yaml`.

### PR DESCRIPTION
**Pull request by chan-hoo**
## DESCRIPTION OF CHANGES: 
- (bug fix) Updates the module files for the AQM tasks on WCOSS2.
- (bug fix) Adds the memory size to the workflow xml file for the verification tasks on WCOSS2.
- (new feature) Updates the module files for WCOSS2 to meet the NCO's requirement that each task should only load the modules necessary for the task itself).
- (new feature) Updates the hash of UFS_UTILS with the latest one as of 5/15/23.

### Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## TESTS CONDUCTED: 
- WE2E tests on WCOSS2 and Hera:
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
MET_verification
- AQM sample `community` run on WCOSS2 and Hera

- [x] hera.intel
- [ ] orion.intel
- [ ] cheyenne.intel
- [ ] cheyenne.gnu
- [ ] gaea.intel
- [ ] jet.intel
- [x] wcoss2.intel
- [ ] NOAA Cloud (indicate which platform)
- [ ] Jenkins
- [ ] fundamental test suite
- [ ] comprehensive tests (specify *which* if a subset was used)

## ISSUE: 
- Fixes issue mentioned in #789 
- Fixes issue mentioned in #785 

## CHECKLIST
- [x] My code follows the style guidelines in the Contributor's Guide
- [x] I have performed a self-review of my own code using the Code Reviewer's Guide
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes do not require updates to the documentation (explain).
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

File: modulefiles/tasks/wcoss2/nexus_gfs_sfc.local.lua [hunk 1]
```diff
@@ -0,0 +1 @@
+load("python_srw")
```
Comment (christinaholtNOAA) [inline modulefiles/tasks/wcoss2/nexus_gfs_sfc.local.lua]: If this is the only thing it does, that is no longer necessary. This one should be loaded by default.

File: ush/load_modules_run_task.sh [hunk 2]
```diff
@@ -99,41 +99,16 @@ set -u
...
-    module load mod_aqm-utils
-  elif [ "${task_name}" = "nexus_emission" ] || ([ "${task_name}" = "nexus_post_split" ] && \
-       [ "${machine}" = "wcoss2" ] ); then
-    module load mod_nexus
-  fi
-else
-  module use "${HOMEdir}/modulefiles"
-  module load "${BUILD_MOD_FN}" || print_err_msg_exit "\
-  Loading of platform- and compiler-specific module file (BUILD_MOD_FN) 
+if [ "${WORKFLOW_MANAGER}" = "rocoto" ]; then
```
Comment (christinaholtNOAA) [inline ush/load_modules_run_task.sh]: This logic doesn't seem to be related to whether it's just for Rocoto. We could also use this file to load modules even if we were running from wrapper scripts. Please use less broad logic to make this choice.

**Review by christinaholtNOAA — APPROVED** [review]
@chan-hoo I'm sorry again that it was one of my contributions that broke AQM on WCOSS. We are in desperate need of tests that protect these capabilities!

I left a couple of super tiny items below. Once the logic one is addressed, everything looks good to me.

File: ush/load_modules_run_task.sh (hunk 2 above)
Comment (chan-hoo) [reply ush/load_modules_run_task.sh]: Removed.

File: modulefiles/tasks/wcoss2/nexus_gfs_sfc.local.lua (hunk 1 above)
Comment (chan-hoo) [reply modulefiles/tasks/wcoss2/nexus_gfs_sfc.local.lua]: Removed.

File: ush/config.aqm.nco.realtime.yaml [hunk 3]
```diff
@@ -41,6 +43,7 @@ rocoto:
     metatask_run_ensemble:
       task_run_fcst_mem#mem#:
         walltime: 04:00:00
+#    task_aqm_ics_ext:
```
Comment (MichaelLueken) [inline ush/config.aqm.nco.realtime.yaml:46]: Should this commented line (46) be removed?

**Review by MichaelLueken — APPROVED** [review]
@chan-hoo I was able to successfully run the `ush/config.aqm.community.yaml` configuration on Hera without issue.  Outside of a minor issue in `ush/config.aqm.nco.realtime.yaml`, I'm fine with these changes.

Since the automated Jenkins tests don't test on WCOSS2 and don't exercise the `ush/config.aqm.nco.realtime.yaml` file, I will approve these changes and submit the tests now.

**Review by MichaelLueken — APPROVED** [review]
@chan-hoo I was able to successfully run the `ush/config.aqm.community.yaml` configuration on Hera without issue.  Outside of a minor issue in `ush/config.aqm.nco.realtime.yaml`, I'm fine with these changes.

Since the automated Jenkins tests don't test on WCOSS2 and don't exercise the `ush/config.aqm.nco.realtime.yaml` file, I will approve these changes and submit the tests now.

File: ush/config.aqm.nco.realtime.yaml (hunk 3 above)
Comment (chan-hoo) [reply ush/config.aqm.nco.realtime.yaml:46]: With a new workflow xml structure, it is not easy to exclude 'task_aqm_ics_ext' for a cold start in AQM. Previously, it was controlled in the template of the xml file with the parameter 'coldstart'. In a new system, the only way is to add this line to the configuration file. So, users are able to test a cold start just by uncommenting this line. In my opinion, this line is important in this sample script.

**Comment by chan-hoo** [comment]
@MichaelLueken, thank you for your approval!

File: ush/config.aqm.nco.realtime.yaml (hunk 3 above)
Comment (MichaelLueken) [reply ush/config.aqm.nco.realtime.yaml:46]: Understood.  Resolving conversation now.

**Comment by MichaelLueken** [comment]
@chan-hoo The Jenkins tests have completed.  Unfortunately, a majority of tests ultimately failed in the `make_grid` task.  An issue was introduced into the `UFS_UTILS` repo that causes failures in the `make_grid` task in hash [6af8f1a](https://github.com/ufs-community/UFS_UTILS/commit/6af8f1a6ffcec68d88a4f59be633fa443056fe20).  The issue was ultimately identified and corrected in hash [4d2768c](https://github.com/ufs-community/UFS_UTILS/commit/4d2768ced9edf73472a1d23328a2c34a52c3ddfa) last Friday.

Please either update the `UFS_UTILS` hash to [4d2768c](https://github.com/ufs-community/UFS_UTILS/commit/4d2768ced9edf73472a1d23328a2c34a52c3ddfa), or remove the updated hash in the Externals.cfg file.  I am in the process of opening a PR to update the WM, UPP, and UFS_UTILS hashes (as well as comment out the tests that are still failing in the coverage tests and updating the Gaea modulefile so that the SRW will once again compile and run on the machine).  I can remove the UFS_UTILS updated hash from my branch if you wish to update it in this PR.

Please let me know if you have any questions or concerns.  Thanks!

**Comment by chan-hoo** [comment]
@MichaelLueken, updated to `4d2768c`.

**Comment by MichaelLueken** [comment]
Following the update to the UFS_UTILS hash, the `nco_grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15_thompson_mynn_lam3km` test failed on Cheyenne GNU and the `grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_RRFS_v1beta` test failed on Hera Intel.  These two issues are expected failures.  Until the modulefile for Gaea is updated (PR #799), the tests will fail for that machine as well.

However, there was an unexpected failure on Hera GNU.  The 
 `nco_grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16` test failed.  A rerun of the Jenkins tests for Hera GNU showed that all tests passed (the green checkmark for the `continuous-integration/jenkins/pr-merge` check shows that the tests all passed for Hera GNU).

Moving forward with merging this work now.
